ARCHITECTURE.md

```diff
@@ -0,0 +1,152 @@
+![BluepillIcon](doc/img/bluepill_text.png)
+
+![](https://github.com/linkedin/bluepill/workflows/master/badge.svg)
+
+## iOS Simulator Architecture
+
+- App (builds with) -> (Xcode/Bazel) -> `X.app` (packaged into an IPA, an archive of the app or "Bundle")
+- iPhones have ARM processors while Macs have `x86_64` (pre M1 at least)
+- Compilation results in an architecture specific binary (or app binary)
+
+## Bluepill Build How-To
+- `bluepill.sh` builds Bluepill in CI (It has a sample app)
+- Technically the script builds Bluepill & the sample app, and runs the tests
+- `BPSampleApp.app` is the "bundle", an x86_64 bundle but iPhone needs an ARM executable
```

Comment:
BPSampleApp is the test bed we use for testing Bluepill functionalities. It contains test targets that simulates real world scenarios, e.g a test that crashes / timeout / blocks UI, etc.